File: 프로그래머스/lv2/12949. 행렬의 곱셈/행렬의 곱셈.py
def solution(arr1, arr2):
    rn1, rn2 = len(arr1), len(arr2)
    cn1, cn2 = len(arr1[0]), len(arr2[0]) 
    answer = [[0]*cn2 for _ in range(rn1)]
    
    for i in range(rn1):
        for k in range(cn2):
            sum = 0
            for j in range(cn1):
                sum += arr1[i][j]*arr2[j][k]
            answer[i][k] = sum
            
    return answer

File: 프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.java
class Solution {
    public int[][] solution(int[][] arr1, int[][] arr2) {
        int[][] answer = new int[arr1.length][arr2[0].length];
        int e;
        for(int i=0; i<arr1.length; i++){
            
            for(int k=0; k<arr2[0].length; k++){
                e = 0;
                for(int j=0; j<arr1[0].length; j++){
                    e += arr1[i][j] * arr2[j][k];
                }
                answer[i][k] = e;
            }
        }
        return answer;
    }
}